### DISCUSSION TITLE
Full backup

### DISCUSSION THREAD

**Discussion by umpfmpf** [devemux86/cruiser#528] [Q&A] [answered]
Would it be a way to add the option to back up all private settings and data at once?
Maybe even in a folder or as a complete backup in a ZIP file? And then, when playing back the backup, select which data should be imported.
And this file could also be copied to a 2. Device to synchronize the data and settings.

**Comment by devemux86** [comment]
@umpfmpf thanks for the suggestion.

It's an interesting idea, we can think about it.

**Comment by ramondevic** [comment]
> Would it be a way to add the option to back up all private settings and data at once? Maybe even in a folder or as a complete backup in a ZIP file? And then, when playing back the backup, select which data should be imported. And this file could also be copied to a 2. Device to synchronize the data and settings.

That's a GREAT idea :-)

**Comment by Ciutnik** [comment]
It's true that being able to make copies - especially in the cloud - would be nice.

I like the "Backup Manager" used in Locus. A ZIP is created and can be automatically sent to the cloud. You decide what will be included in the backup (data and/or settings), whether the copies should be created automatically or manually, as well as what you want to restore from the backup (data and/or settings).

**Comment by bios4** [comment]
Not directly related with the very export option in this thread, but related to import/export in general: currently when e.g. settings are exported via Main > Profile > three-dot-menu > Export there is no information if everything worked well or if there was an issue during the process....

After a name is chosen for the export zip, nothing more "happens" - Cruiser just returns to the Profile screen without any further notice.

I would really appreciate if you could implement a short toast message after successful importing or exporting of settings @devemux86, [like you did recently](https://github.com/devemux86/cruiser/discussions/516#discussioncomment-9251670) for the saving of favorites.

The same applies of course when there went something wrong... 😉

**Reply by devemux86** [reply]
Have you ever seen any export to not succeed?

It also depends on what the Android allows the apps to know about the save.

**Comment by bios4** [comment]
> Have you ever seen any export to not succeed?

Not personally, but obviously there went [SOMETHING wrong](https://github.com/devemux86/cruiser/discussions/527#discussioncomment-9328392) during the export of the data of DolomitesTom...

I also can't quite figure out your counter-question - is it possible/not possible/too time-consuming/... to display a corresponding message?

**Reply by devemux86** [reply]
That report was not from the export initiated by the menu.
Something else went wrong, but unfortunately it couldn't be reproduced by either of us.

About the question, if Android export can report success or not, then we can show a message.
If not, displaying always a success message is not very useful.

**Reply by bios4** [reply]
I see.

> If not, displaying always a success message is not very useful.

Of course a success message should not ALWAYS be displayed but only if a success can be determined...

**Reply by devemux86** [reply]
I will check what is possible, everything depends on how Android processes work.

**Comment by HPK-AT** [comment]
Hi. I want to wake up this threat from  umpfmpf again.
Export / Import all Settings and chooseable while restore which Parts to Import.
Here is a screenshot how it could be integrated.
And no , it does not have to be in Google Backup integrated,  i would more prefere to choose my own where the data is stored.
 Localy or on Network drive or on Standard Android possibilities like Quickshare, Google Drive and so on
<img src="https://github.com/user-attachments/assets/1a2feeb0-86b9-40cb-bbaf-b0f6e9b051fc" width="50%"/>

**Reply by HannesHoBu** [reply]
I would also like to push this thread.
Additionally I would ask for a backup solution for the recorded tracks. As far as I know they are stored in the file 1.gpx.
I was not able to transfer this file between different devices.

**Reply by devemux86** [reply]
@HannesHoBu The GPS recorded track is stored in app database.

- You can export it and import it in another device using the GPS recording menu.

- You can also save the recorded track in My collection / Tracks,
clear it and start recording a new one.

**Reply by HannesHoBu** [reply]
@devemux86 Emux, many thanks for your quick reply. Obviously I am too blond follow your instructions. I can't find the file with the recorded tracks. In which directory can the relevant file be found?

**Reply by devemux86** [reply]
@HannesHoBu If you want the recorded track as a file, use the menu "**GPS recording**" and select "**Export**" or "**Share**".

**Reply by HannesHoBu** [reply]
I refer to the automatic recorded tracks 
![Screenshot_2025-05-27-22-12-07-565_gr talent cruiser~2](https://github.com/user-attachments/assets/7b7d8073-e5c3-4084-9b8a-b143032e68c8)

**Reply by devemux86** [reply]
@HannesHoBu I also mean the automatic recorded tracks, we are discussing the same thing.

Use the menu "**GPS recording**" where you can:

- "**Save**" a recorded track in my collection

- "**Import**" a GPX file as recorded track

- "**Export**" or "**Share**" the recorded track as GPX file

**Reply by HannesHoBu** [reply]
Finally I made it! Thanks for your patience and support.
This is a cool feature.

**Comment by umpfmpf** [comment]
Are there any new features regarding a backup function to keep multiple devices synchronized?

**Reply by devemux86** [reply]
You can use any free cloud service like Google Drive, Dropbox, etc. and synchronize your backups.

**Reply by umpfmpf** [reply]
But it is not possible to create a backup with all settings and saved routes and poi's and send it to a 2nd device without cloud?

**Reply by devemux86** [reply]
You are asking for a zip file with all the exports included.

It is the same as syncing multiple files, there is no difference.

**Reply by WolfgangZ42** [reply]
> But it is not possible to create a backup with all settings and saved routes and poi's

My workflow is to export **all profiles**, **all favorites**, **all routes** and **all tracks** on my smartphone (_for navigation_) directly into a Dropbox folder. And to synchronize my tablet (_for planning_) I import these 4 files from this Dropbox folder. That's it. 
Here the files of my last synch:

![2026-03-02 11_17_48-D__Dropbox_Cruiser](https://github.com/user-attachments/assets/089b3dc7-f3b1-4086-a691-1f2894813548)

**Reply by MotoUKRider** [reply]
I think jist of the request is for Cruiser to do _all of that_ in one action _for daft motorcyclists who do daft things even when not on a motorcycle_ :clown_face: 

It's a good example of a Wiki topic, though!  The result shown by @WolfgangZ42 works fine, but there are several menus/places you need to visit to produce file backups like this.  Documenting the workflow would be helpful.

**Comment by HPK-AT** [comment]
as mentioned here: https://github.com/devemux86/cruiser/discussions/528#discussioncomment-13155969 
it would be nice to have just one export/import button and you can choose what to import or export

**Reply by devemux86** [reply]
I will see it.

**Comment by devemux86** [comment]
Full backup and restore with a single file process and data selection will be available for testing in the next Cruiser Beta.

<img src="https://github.com/user-attachments/assets/e41cfda7-0e8c-49e7-b1fa-ab21e20f24af" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/901

Export and import all data using the app profile menu.

**Comment by freeasabeer** [comment]
I come after the battle and I am very sorry for that, but are the app settings also exported ?
As far as I understood what was implemented here, I believe the answer is no.
Is there another way to export them which I would have missed ? If not, would it be feasible to implement the export of this additional set of data ?

**Reply by devemux86** [reply]
@freeasabeer The app settings are included in the **profiles**, so that they can also be exported and imported.

**Comment by Leeb64** [comment]
Meckermodus an->  Jemanden der nicht weiß wie man ein Backup erstellt und nicht durch Zufall darüber stolpert,der sucht unter Umständen ewig dannach.
Könnte man nicht in "Einstellungen" eventuell unter "Applikation" diese Backup-Funktion integrieren?

**Reply by devemux86** [reply]
> Jemanden der nicht weiß wie man ein Backup erstellt und nicht durch Zufall darüber stolpert,der sucht unter Umständen ewig dannach.
Könnte man nicht in "Einstellungen" eventuell unter "Applikation" diese Backup-Funktion integrieren?

Good idea, this will be available in the next update, as an additional place to start the backup.

**Reply by bios4** [reply]
> Good idea

As already suggested a long time a go [here](https://github.com/devemux86/cruiser/discussions/898#discussioncomment-16253508) and [here](https://github.com/devemux86/cruiser/discussions/898#discussioncomment-16462347), but w/o any response or reaction to it so far.

But finally good to see that this is considered...

**Reply by devemux86** [reply]
Post **edits** do not generate email notifications, so they are usually never read.

For new content and ideas, it is best to create new posts.

**Reply by devemux86** [reply]
Backup will be available in the application settings:

<img src="https://github.com/user-attachments/assets/5f8b2148-1258-4c53-aee4-6e5568d96b36" width="50%"/><img src="https://github.com/user-attachments/assets/acdc3d21-ca2b-4ea5-abd7-0f371430bb6c" width="50%"/>

**Reply by Leeb64** [reply]
exactly like that.

**Reply by kds112** [reply]
An additional Restore function would be useful at this position as well. With a dialog whether you what to replace or merge the data with the backupped data.

**Reply by WolfgangZ42** [reply]
> replace or merge the data with the backupped data

That's an interesting idea - it's happened to me a few times that backups were merged even though I didn't want them to be.

**Reply by kds112** [reply]
After selecting a backup file Cruiser should analyse what data is inside the file (tracks, favs, etc.) and offers the selection to replace and/or merge the data in Cruiser and the backup file.

**Reply by devemux86** [reply]
> An additional Restore function would be useful at this position as well.

> With a dialog whether you what to replace or merge the data with the backup data.

We can do these:

<img src="https://github.com/user-attachments/assets/d9190769-4523-4a43-bf8d-5ccbe8bad396" width="50%"/><img src="https://github.com/user-attachments/assets/3584ff64-bea0-4a8d-b2d6-486369609747" width="50%"/>

**Reply by kds112** [reply]
Import should offer only the data types that are part of the backup file (or is it too complicated?)
I would Insert a space or line between Profiles and Replace. Is the function simply clear for a new user? 🤔

**Reply by devemux86** [reply]
> Import should offer only the data types that are part of the backup file (or is it too complicated?)

It is not possible to check the content of files before import because of Android storage limitations.

**Reply by devemux86** [reply]
>  I would Insert a space or line between Profiles and Replace. Is the function simply clear for a new user? 🤔

Replace is a "dangerous" action. It's best to not offer it?
Users can delete what they do not want before or after import.

**Reply by bios4** [reply]
> Users can delete what they do not want before or after import.

This is only reasonable for small collections of items!

Even if replace seems "dangerous": it is the responsibility of the user, right!??

**Reply by bios4** [reply]
Why not offer two "Import" menu items:

- "Import & merge"
- "Import & replace"

...and show the screen with checkboxes only after the selection!?

**Reply by kds112** [reply]
Well, I would offer the replace function with a security query "Are you sure to replace the internal data?"

**Reply by devemux86** [reply]
> Why not offer two "Import" menu items

This is a better solution, I updated the screenshots [above](https://github.com/devemux86/cruiser/discussions/528#discussioncomment-16602123).

**Reply by kds112** [reply]
Merge is the more suitable verb than Update. Will there be dialog "Are you really sure to replace..."?

**Reply by devemux86** [reply]
There is already a 2nd confirmation dialog with check boxes for what to import.
(see [above](https://github.com/devemux86/cruiser/discussions/528#discussioncomment-16602123))

**Comment by devemux86** [comment]
Backup has been moved to "Settings | Application" with a replace option.

You can test it with **Cruiser 5.9.2 Beta**:
- https://github.com/devemux86/cruiser/discussions/898#discussioncomment-16611359

**Reply by DolomitesTom** [reply]
Tested export on smartphone and restore on tablet. For me it's working.
